codalab-service.py

```diff
@@ -0,0 +1,387 @@
+
+    def push_image(self, image):
+        self._run_docker_cmd('push codalab/%s:%s' % (image, self.args.version))
+
+    def _run_compose_cmd(self, cmd):
+        files_string = ' -f '.join(self.compose_files)
+        command_string = 'docker-compose -p %s -f %s %s' % (self.instance_name, files_string, cmd)
+        subprocess.check_call(command_string, cwd=self.compose_cwd, env=self.compose_env, shell=True)
+
+    def bring_up_service(self, service):
+        self._run_compose_cmd('up -d --no-deps --no-recreate %s' % service)
+
+    def run_service_cmd(self, cmd, root=False, service='rest-server'):
+        if root:
+            uid = '0:0'
+        else:
+            uid = self.compose_env['CODALAB_UID']
+        self._run_compose_cmd('run --no-deps --rm --entrypoint="" --user=%s ' % uid + service + ' bash -c "%s"' % cmd)
+
+    def start_service(self):
+        print("[CODALAB] ==> Starting MySQL")
+        self.bring_up_service('mysql')
+
+        print("[CODALAB] ==> Configuring the service")
+        self.run_service_cmd("/data/bin/wait-for-it.sh mysql:3306 -- /opt/codalab-worksheets/codalab/bin/cl config server/engine_url mysql://%s:%s@mysql:3306/codalab_bundles && /opt/codalab-worksheets/codalab/bin/cl config cli/default_address http://rest-server:2900 && /opt/codalab-worksheets/codalab/bin/cl config server/rest_host 0.0.0.0" % (self.compose_env['CODALAB_MYSQL_USER'], self.compose_env['CODALAB_MYSQL_PWD']), root=(not self.args.service_home))
+
+        if self.args.initial_config:
+            print("[CODALAB] ==> Creating root user")
+            self.run_service_cmd("/opt/codalab-worksheets/venv/bin/pip install /opt/codalab-worksheets && /data/bin/wait-for-it.sh mysql:3306 -- /opt/codalab-worksheets/venv/bin/python /opt/codalab-worksheets/scripts/create-root-user.py %s" % self.compose_env['CODALAB_ROOT_PWD'], root=True)
+
+        print("[CODALAB] ==> Starting rest server")
+        self.bring_up_service('rest-server')
+
+        if self.args.initial_config:
+            print("[CODALAB] ==> Creating initial worksheets")
+            self.run_service_cmd("/data/bin/wait-for-it.sh rest-server:2900 -- opt/codalab-worksheets/codalab/bin/cl logout && /opt/codalab-worksheets/codalab/bin/cl new home && /opt/codalab-worksheets/codalab/bin/cl new dashboard", root=(not self.args.service_home))
+
+        print("[CODALAB] ==> Starting bundle manager")
+        self.bring_up_service('bundle-manager')
+        print("[CODALAB] ==> Starting frontend")
+        self.bring_up_service('frontend')
+        print("[CODALAB] ==> Starting nginx")
+        self.bring_up_service('nginx')
+        if self.args.start_worker:
+            print("[CODALAB] ==> Starting worker")
+            self.bring_up_service('worker')
+
+    def build(self):
+        print("[CODALAB] => Building Docker images")
+        IMAGE_TO_SUFFIX = {
```

Comment:
can this just be the identity map?

Reply:
Yep, I guess there's no harm to renaming our Dockerfiles, which I was trying to avoid.